internal/rickandmorty/util.go

```diff
@@ -0,0 +1,139 @@
+package rickandmorty
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io"
+	"reflect"
+	"strconv"
+)
+
+var (
+	ErrInvalidCodecFormat = errors.New("invalid codec format")
+	ErrEmptyCSV           = errors.New("empty csv")
+	ErrDecodeInput        = errors.New("cannot decode input")
+)
+
+type CharacterCodec interface {
+	Encode(writer io.Writer, data interface{}) error
+	Decode(reader io.Reader, data interface{}) error
+}
+
+type jsonCodec struct{}
+
+func NewJSONCharacterCodec() CharacterCodec {
+	return jsonCodec{}
+}
+func (j jsonCodec) Encode(writer io.Writer, data interface{}) error {
+	switch data.(type) {
+	case Character, CharacterList:
+		return json.NewEncoder(writer).Encode(data)
+	default:
+		return ErrInvalidCodecFormat
+	}
+}
```

Comment:
Why not use just an if for the first return and for the second one just return outside if